### PR TITLE
Update potential_flow_test_factory.py

### PR DESCRIPTION
**Pull request by philbucher**
after #7182 
I also used some newer functionalities for skipping tests

Plus some TODOs for you to improve the tests

(I didn't run the tests, please check yourself)

BTW will you rename the app to `PotentialFlowApplication`? I thought this was planned?

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/tests/potential_flow_test_factory.py [hunk 1]
```diff
@@ -42,11 +36,8 @@ def setUp(self):
         # Set to true to get post-process files for the test
         self.print_output = False
 
+    @UnitTest.skipIfApplicationsNotAvailable("ExternalSolversApplication", "HDF5Application")
```
Comment (marcnunezc) [inline applications/CompressiblePotentialFlowApplication/tests/potential_flow_test_factory.py:39]: Can we also start replacing the ExternalSolvers used in the test with the LinearSolvers? Or is it better to wait?

**Review by marcnunezc — APPROVED** [review]
Ok thanks!

Regarding the name-change, I personally prefer `PotentialFlowApplication`. My only concern if it is worth losing the commit history for each file (maybe Im wrong and the history is not lost) because of just the name change. Also, would it greatly affect the .git history size? What is your opinion on this?

Thinking about it again, losing the history shouldn't be a big worry if this app is meant to be merged to the Fluid in the future, as the file-history would be lost anyway.

File: applications/CompressiblePotentialFlowApplication/tests/potential_flow_test_factory.py (hunk 1 above)
Comment (philbucher) [reply applications/CompressiblePotentialFlowApplication/tests/potential_flow_test_factory.py:39]: yes!
no there will be no more interface changes

**Comment by philbucher** [comment]
> Regarding the name-change, I personally prefer PotentialFlowApplication.

thats what I wrote isn't it?

The history is not lost, git recognizes the rename. However it is no longer available through the github interface, only through the terminal (at least was like this last time I checked)

**Comment by marcnunezc** [comment]
> > Regarding the name-change, I personally prefer PotentialFlowApplication.
> 
> thats what I wrote isn't it?

Yeah, sorry! It was not a message trying to contradict your comment. I meant that I also personally would prefer the name `PotentialFlowApplication` rather than keeping the current one. 

> The history is not lost, git recognizes the rename. However it is no longer available through the github interface, only through the terminal (at least was like this last time I checked)

I think we can discard the history being an issue for the name-change then. 

The only issue that I can think of then is the fact that old-scripts would stop working, which should not be a big problem. What do you think @inigolcanalejo? I see this was part of the to-do list in #4533

**Comment by inigolcanalejo** [comment]
I also prefer the name `PotentialFlowApplication`. Would it be ok to make the change after we have a fist 3d version running? I have some scripts that work with old branches and it would be inconvenient having to duplicate them.

**Pull request merged**